src/passes/LocalCSE.cpp

```diff
@@ -63,7 +63,7 @@ struct LocalCSE : public WalkerPass<LinearExecutionWalker<LocalCSE>> {
 
   struct UsableHasher {
     HashType operator()(const Usable value) const {
-      return rehash(value.hashed.hash, value.localType.getID());
+      return rehash(uint64_t(value.hashed.hash), value.localType.getID());
```

Comment:
this isn't obvious to me, what's going on here?

Reply:
`value.hashed.hash` is a uint32_t and `value.localType.getID` is now a uint64_t, so this became an ambiguous overload of `rehash`. The cast makes it unambiguous again.

Reply:
Was it a compiler error that led to the fix?

What concerns me is that this is just a random pass, so it's not obvious to me why this pass and not others needed such a fix.

Reply:
Yeah it was a compiler error about the ambiguous overload.